packages/js/src/cache/notifications-cache.ts

```diff
@@ -0,0 +1,253 @@
+import { InMemoryCache } from './in-memory-cache';
+import type { Cache } from './types';
+import type { ListNotificationsArgs, ListNotificationsResponse, Notification } from '../notifications';
+import { NotificationEvents, NotificationsEvents, NovuEventEmitter } from '../event-emitter';
+import type { NotificationFilter } from '../types';
+import { areTagsEqual, isSameFilter } from '../utils/notification-utils';
+
+const getCacheKey = (args: ListNotificationsArgs): string => {
+  return JSON.stringify(args);
```

Comment:
⚠️ I think we should be doing `{ tags, read, archived }` because `JSON.stringify` preserves the order of the properties. This means that we could get different results for the same filter.

Reply:
good catch!